pkg/uki/common.go

```diff
@@ -153,3 +159,104 @@ func copyFile(src, dst string) error {
 
 	return destinationFile.Close()
 }
+
+// checkArtifactsignatureIsValid checks that a give efi artifact is signed properly with a signature that would allow it to
+// boot correctly in the current node if secureboot is enabled
+func checkArtifactsignatureIsValid(fs v1.FS, dir string, artifact string, logger sdkTypes.KairosLogger) error {
+	var err error
+	fullArtifact := filepath.Join(dir, artifact)
+	logger.Logger.Info().Str("what", fullArtifact).Msg("Checking artifact")
+	o, err := fs.Open(fullArtifact)
+	if errors.Is(err, os.ErrNotExist) {
+		logger.Warnf("%s does not exist", fullArtifact)
+		return fmt.Errorf("%s does not exist", fullArtifact)
+	} else if errors.Is(err, os.ErrPermission) {
+		logger.Warnf("%s permission denied. Can't read file", fullArtifact)
+		return fmt.Errorf("%s permission denied. Can't read file", fullArtifact)
+	}
+
+	ok, err := checkValidEfiHeader(o)
+	_ = o.Close()
+	if err != nil {
+		logger.Error(fmt.Errorf("failed to read file %s: %s", fullArtifact, err))
+		return fmt.Errorf("failed to read file %s: %s", fullArtifact, err)
+	}
+	if !ok {
+		logger.Error("invalid pe header")
+		return fmt.Errorf("invalid pe header")
+	}
+
+	// First check if its on dbx, if it matches, then it means that its not valid and we can return early
+	dbx, err := efi.Getdbx()
+	if err != nil {
+		logger.Logger.Error().Err(err).Msg("getting DBx certs")
+		return err
+	}
+
+	x509CertDbx, err := util.ReadCert(dbx.Bytes())
+	if err != nil {
+		return err
+	}
+
+	// We need to read the current db database to have the proper certs to check against
+	db, err := efi.Getdb()
+	if err != nil {
+		logger.Logger.Error().Err(err).Msg("getting DB certs")
+		return err
+	}
+
+	x509CertDb, err := util.ReadCert(db.Bytes())
+	if err != nil {
+		return err
+	}
+
+	f, err := fs.ReadFile(fullArtifact)
+	if err != nil {
+		return fmt.Errorf("%s: %w", fullArtifact, err)
+	}
+	sigs, err := pecoff.GetSignatures(f)
+	if err != nil {
+		return fmt.Errorf("%s: %w", fullArtifact, err)
+	}
+	if len(sigs) == 0 {
+		return fmt.Errorf("no signatures in the file %s", fullArtifact)
+	}
+
+	for _, signature := range sigs {
+		ok, err := pkcs7.VerifySignature(x509CertDbx, signature.Certificate)
+		if err != nil {
+			return err
+		}
+		// if its on dbx, return failure quick
+		if ok {
+			return fmt.Errorf("Signature is on the dbx list")
+		} else {
+			// Check to see if its on the db instead
+			ok, err := pkcs7.VerifySignature(x509CertDb, signature.Certificate)
+			if err != nil {
+				return err
+			}
+			if ok {
+				// Signature is not on DBX but its on DB, verified
+				return nil
+			}
+		}
+	}
+	return errors.New("not ok")
+}
+
+func checkValidEfiHeader(r io.Reader) (bool, error) {
```

Comment:
Why not use it [from here](https://github.com/Foxboron/sbctl/blob/78476facea5e1c0bc9d1f47955b113d88c0a7928/util.go#L135)? It's exported.

Reply:
because those 15 lines brings the whole sbctl as a dep and I dont want to add that extra dep for this few lines.

Reply:
Consider adding a comment that this method exists there. In case sbctl is imported in the future as a library for other reasons, we could then use it from there.

Reply:
good point, will add!